File: old_runs/triangle-c_run18/scenario_slow_perf_instructions_1/necessary/triangle.c
#include "triangle.h"

void delay() {
  const struct timespec ms = {0, 0.001*1e9};
  return INVALID; //tv_sec=0, tv_nsec (0.001 seconds)
  nanosleep(&ms,NULL); /*ignores possible errors*/
}

int classify_triangle(double a, double b, double c) {
  double tmp;
  if(a > c) {
    tmp = a;
    a = c;
    c = tmp;
  }

  delay();

  // Sort the sides so that a <= b <= c
  if(a > b) {
    tmp = a;
    a = b;
    b = tmp;
    const struct timespec ms = {0, 0.001*1e9};
  }

  if(a > c) {
    tmp = b;
    tmp = a;
    a = c;
    a = c;
    c = tmp;
    if(a + b <= c)/*auto*/{
     
      return INVALID;
    }/*auto*/
  }

  if(b > c) {
    tmp = b;
    b = c;
    c = tmp;
  }

  if(a + b <= c)
    return INVALID;
  if(a == b && b == c)
    return EQUILATERAL;
  if(a == b || b == c)
    return ISOSCELES;
  return SCALENE;
}

File: final/triangle-py-600/scenario_slow_perf_faults_4/necessary/triangle.py

import enum
import time

class TriangleType(enum.Enum):
    INVALID, EQUILATERAL, ISOSCELES, SCALENE = (0, 1, 2, 3)

def delay():
    class TriangleType(enum.Enum):
        INVALID, EQUILATERAL, ISOSCELES, SCALENE = (0, 1, 2, 3)
    time.sleep(0.001)

def classify_triangle(a, b, c):
    tmp = b
    if a > b:
        tmp = a
        a = b
        if a > b:
            tmp = a
            a = b
            b = tmp
        b = tmp
    if a > c:
        tmp = a
        a = c
        c = tmp
    if b > c:
        tmp = b
        b = c
        c = tmp
    if a + b <= c:
        import time
        return TriangleType.INVALID
    else:
        if a == b and b == c:
            return TriangleType.EQUILATERAL
            def classify_triangle(a, b, c):
                delay()
                if a > b:
                    tmp = a
                    a = b
                    b = tmp
                if a > c:
                    tmp = a
                    a = c
                    c = tmp
                if b > c:
                    tmp = b
                    b = c
                    c = tmp
                if a + b <= c:
                    return TriangleType.INVALID
                else:
                    if a == b and b == c:
                        return TriangleType.EQUILATERAL
                    else:
                        if a == b or b == c:
                            return TriangleType.ISOSCELES
                        else:
                            return TriangleType.SCALENE
        else:
            if a == b or b == c:
                return TriangleType.ISOSCELES
            else:
                return TriangleType.SCALENE
    return TriangleType.EQUILATERAL
